### PR TITLE
Enable opening external urls in browsers

### PR DESCRIPTION
**Pull request by dianabarsan**
medic/cht-core#6648

### CODE REVIEW

**Comment by dianabarsan** [comment]
@garethbowen please have a look. Thanks!

File: src/main/java/org/medicmobile/webapp/mobile/EmbeddedBrowserActivity.java [hunk 1]
```diff
@@ -369,8 +369,9 @@ private void enableStorage(XWalkView container) {
 	private void enableUrlHandlers(XWalkView container) {
 		container.setResourceClient(new XWalkResourceClient(container) {
 			@Override public boolean shouldOverrideUrlLoading(XWalkView view, String url) {
-				if(url.startsWith("tel:") || url.startsWith("sms:")) {
-					Intent i = new Intent(Intent.ACTION_VIEW, Uri.parse(url));
+				Uri uri = Uri.parse(url);
+				if(url.startsWith("tel:") || url.startsWith("sms:") || !isUrlRelated(appUrl, uri)) {
```
Comment (garethbowen) [inline src/main/java/org/medicmobile/webapp/mobile/EmbeddedBrowserActivity.java:373]: I think we can just call `isUrlRelated` and drop the startsWith calls, eg:

```suggestion
				if(!isUrlRelated(appUrl, uri)) {
```

Then, maybe it makes sense to invert the `if` so you're asking a positive.

As a nice to have, I see that `isUrlRelated` takes a `Uri` but immediately `toStrings` it - would it be worth adding a variant which takes a string so we don't have to parse it most of the time?

Finally, we actually have a few tests for the Utils so if you do the above checks it'd be worth adding unit tests to assert that `sms` and `tel` and `mailto` links fire the intent.

**Review by garethbowen — CHANGES_REQUESTED** [review]
Nice one!

A few clean up suggestions for discussion...

File: src/main/java/org/medicmobile/webapp/mobile/EmbeddedBrowserActivity.java (hunk 1 above)
Comment (dianabarsan) [reply src/main/java/org/medicmobile/webapp/mobile/EmbeddedBrowserActivity.java]: Done!

**Review by garethbowen — APPROVED** [review]
Great work!

**Pull request merged**